HLTrigger/Egamma/plugins/HLTPMMassFilter.cc

```diff
@@ -134,38 +129,37 @@ bool HLTPMMassFilter::hltFilter(edm::Event& iEvent,
 
       TLorentzVector pscEle = approxMomAtVtx(theMagField, vertexPos, sc, -1);
       pEleCh2.push_back(pscEle);
-      etaOrig.push_back(sc->eta());
     }
 
+    std::vector<bool> save_cand(scs.size(), true);
     for (unsigned int jj = 0; jj < scs.size(); jj++) {
-      TLorentzVector p1Ele = pEleCh1.at(jj);
-      for (unsigned int ii = 0; ii < scs.size(); ii++) {
-        TLorentzVector p2Ele = pEleCh2.at(ii);
-
-        if (fabs(p1Ele.E() - p2Ele.E()) < 0.00001)
-          continue;
-
-        TLorentzVector pTot = p1Ele + p2Ele;
-        double mass = pTot.M();
-
-        if (mass >= lowerMassCut_ && mass <= upperMassCut_) {
+      for (unsigned int ii = jj + 1; ii < scs.size(); ii++) {
+        if (isGoodPair(pEleCh1[jj], pEleCh2[ii]) or isGoodPair(pEleCh2[jj], pEleCh1[ii])) {
           n++;
-          refsc = scs[ii];
-          filterproduct.addObject(TriggerCluster, refsc);
-          refsc = scs[jj];
-          filterproduct.addObject(TriggerCluster, refsc);
+          for (auto const idx : {jj, ii}) {
+            if (save_cand[idx])
+              filterproduct.addObject(TriggerCluster, scs[idx]);
+            save_cand[idx] = false;
+          }
```

Comment:
In the non-`isElectron1_ && isElectron2_` case, there is a change in how candidate pairs are counted, as (1) the case `jj == ii` is explicitly skipped (supposedly it was effectively skipped pre-PR too, because of the requirement on delta-Energy), and (2) a given pair of `TriggerCluster`s is only counted at most once: a given pair is considered 'good' if it passes the selection under the "plus minus" _or_ "minus plus" hypothesis, while pre-PR the pair was counted twice when both hypotheses were passing the selection (because both indices of the nested loop ranged from 0 to size-1). Here too, a given `TriggerCluster` is added at most once to the output product, unlike in the pre-PR implementation.